snippets/puppet_setup.erb

```diff
@@ -0,0 +1,46 @@
+<%#
+kind: snippet
+name: puppet_setup
+description: this snippet will configure the Puppet agent
+%>
+<%
+etc_path = (@host.operatingsystem.family == 'Freebsd') ? '/usr/local/etc/puppet' : '/etc/puppet'
+bin_path = (@host.operatingsystem.family == 'Freebsd') ? '/usr/local/bin' : '/usr/bin'
+%>
+
+<% if @host.operatingsystem.family == 'Debian' -%>
+apt-get update
+apt-get install -y puppet
+<% elsif @host.operatingsystem.family == 'Freebsd' -%>
+pkg install -y puppet
+<% elsif @host.operatingsystem.family == 'Redhat' -%>
+yum -t -y -e 0 install puppet
```

Comment:
I've always wondered why we give these options to yum. Certainly it would be useful to see error output, `-e 0` would suppress most things, no?